engine/runtime-integration-tests/src/test/java/org/enso/compiler/ErrorCompilerTest.java

```diff
@@ -465,6 +465,46 @@ public void illegalForeignBody4() throws Exception {
         ir, Syntax.UnexpectedExpression$.MODULE$, "Unexpected expression", 0, 18);
   }
 
+  @Test
+  public void illegalPrivateVariableDeclaration() throws Exception {
+    var ir = parse("private var = 42");
+    assertSingleSyntaxError(
+        ir, Syntax.UnexpectedExpression$.MODULE$, "Unexpected expression", 0, 16);
+  }
+
+  @Test
+  public void illegalPrivateKeywordUseInType() throws Exception {
+    var ir = parse("""
+        type T
+            private
+        """);
+    assertSingleSyntaxError(
+        ir,
+        Syntax.UnexpectedDeclarationInType$.MODULE$,
+        "Unexpected declaration in the body of a type",
+        11,
+        18);
+  }
+
+  @Test
+  public void illegalPrivateKeywordUseInMethodBody() throws Exception {
+    var ir = parse("""
+        method =
+            private priv_nested_method x = x
+        """);
+    assertSingleSyntaxError(
+        ir, Syntax.UnexpectedExpression$.MODULE$, "Unexpected expression", 13, 45);
```

Comment:
Could we have clearer errors than that? This is really not helpful for end-users.

It would be much better if it could say something like "inner methods cannot be `private`"

Reply:
In this particular case, it is not that simple to have a better error message. In many other places (see for example tests in the parser), the error message is much more descriptive. This is just handled by some fallback method in `TreeToIr` and it is not simple to change that.